packages/safe-apps-sdk/src/txs/index.ts

```diff
@@ -1,43 +1,42 @@
 import { METHODS } from '../communication/methods';
-import { TxServiceModel, SendTransactionsArgs, Communicator, SendTransactionsResponse } from '../types';
+import {
+  GatewayTransactionDetails,
+  SendTransactionsParams,
+  GetTxBySafeTxHashParams,
+  Communicator,
+  SendTransactionsResponse,
+} from '../types';
 
 class TXs {
-  private txServiceUrl: string | null = null;
   private readonly communicator: Communicator;
 
   constructor(communicator: Communicator) {
     this.communicator = communicator;
   }
 
-  async getBySafeTxHash(safeTxHash: string): Promise<TxServiceModel> {
-    if (!this.txServiceUrl) {
-      throw new Error("ENV information hasn't been synced yet or there was an error during the process");
+  async getBySafeTxHash(safeTxHash: string): Promise<GatewayTransactionDetails> {
+    if (!safeTxHash) {
+      throw new Error('Invalid safeTxHash');
     }
 
-    const controller = new AbortController();
-    const options = {
-      method: 'GET',
-      signal: controller.signal,
-    };
-    setTimeout(() => controller.abort(), 10000);
-
     try {
-      const res = await fetch(`${this.txServiceUrl}/transactions/${safeTxHash}`, options);
-      if (res.status !== 200) {
-        throw new Error(
-          "Failed to get the transaction. Either safeTxHash is incorrect or transaction hasn't been indexed by the service yet",
-        );
+      const response = await this.communicator.send<
+        'getTxBySafeTxHash',
+        GetTxBySafeTxHashParams,
+        GatewayTransactionDetails
+      >(METHODS.getTxBySafeTxHash, { safeTxHash });
+
+      if (!response.success) {
+        throw new Error(response.error);
       }
 
-      const json = await res.json();
-
-      return json as TxServiceModel;
+      return response.data;
     } catch (err) {
       throw err;
```

Comment:
Not related to this PR but why catching it in the first place then?

Reply:
I'm sorry, could you please elaborate? I see that the `(!response.success)`check can be moved to `communicator.send` method. Is this what you meant?

Reply:
I meant that it's catching and just re-throwing the same error.